src/memo/memo.ml

```diff
@@ -793,6 +797,11 @@ end = struct
     in
     (* update the output cache with the correct value *)
     let deps = List.rev running_state.deps_so_far.deps_reversed in
+    let res =
+      match res with
+      | Ok a -> Ok a
+      | Error exns -> Error (Value.Async (Exn_with_backtrace.Set.of_list exns))
```

Comment:
For a moment I got confused because I didn't understand where we filter out duplicate errors. I guess this happens here, when we turn the list that has duplicates into a set. Perhaps, it's worth adding a comment here that this is what happens here.